ddl/index.go

```diff
@@ -626,6 +626,9 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 		var reorgTp model.ReorgType
 		reorgTp, err = pickBackfillType(w.ctx, job, indexInfo.Unique, d)
 		if err != nil {
+			if !errorIsRetryable(err, job) {
+				job.State = model.JobStateCancelled
```

Comment:
Do we need to return the error here？Or does it feel like this `err` message might be overwritten?